doc/services/retention/index.rst

```diff
@@ -0,0 +1,199 @@
+.. _retention_api:
+
+Retention System
+################
+
+The retention system provides an API which allows applications to read/write
+data from/to memory areas or devices which retain the data whilst the device is
+powered. This is designed for sharing information between different
+applications or for a single application when a device reboots without losing
+state information, the contents of the data should not persist in the event of
+a power failure (or possibly with some low power modes on some devices) nor
+should it be stored to a non-volatile backing store such as :ref:`flash_api`,
+:ref:`eeprom_api` or battery-backed RAM.
+
+The retention system builds on top of the retained data driver, and adds
```

Comment:
Please link to docs that describe `retained data driver`.

Reply:
What I meant was to put the link in the rst.

Reply:
Is my understanding correct that the subsystem is only might not necessarily be compatible with any retained mem driver? For example, this would not work with https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/retained_mem/retained_mem_nrf_gpregret.c, correct?

Reply:
It would work fine with it. You can even see the test here https://github.com/zephyrproject-rtos/zephyr/pull/55125/commits/d7d180b17baf480e3cfd426a98e643acecc88dfd which runs on nRF52